.github/workflows/build.yml

```diff
@@ -28,12 +28,28 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        # macos-13 is x86; macos-14 is ARM
-        os: [ubuntu-24.04, macos-13, macos-14, windows-latest]
-        cargo_flags: [""]
+        build: [linux-x86_64-gnu, linux-aarch64-gnu, macos-x86_64, macos-aarch64, windows-x86_64]
         include:
-        - os: ubuntu-24.04
+        - build: linux-x86_64-gnu
+          os: ubuntu-24.04
+          target: x86_64-unknown-linux-gnu
           cargo_flags: "--all-features"
+        - build: linux-aarch64-gnu
+          os: ubuntu-24.04-arm
+          target: aarch64-unknown-linux-gnu
+          cargo_flags: "--all-features"
+        - build: macos-x86_64
+          os: macos-14
+          target: x86_64-apple-darwin
+          cargo_flags: "--features vendored-openssl"
```

Comment:
What makes this necessary? Does GitHub Actions ship with a Homebrew‐installed OpenSSL by default or something? (macOS doesn’t have a system OpenSSL that you can actually use without hacks, so I’m wondering about the x86‐64 vs. AArch64 difference here.)

It might be possible to use `/usr/local/bin/brew` to install an x86‐64 OpenSSL, depending on how the GitHub Actions environment is set up.

Reply:
Basically, there's no x86_64 library to link against on an M1 host, so vendoring it is just the easiest workaround. I actually don't know if OpenSSL comes from an ARM-based brew install, or if the `openssl` crate does some funky shit in order to work with the system-provided libressl, to be fair.

Reply:
It looks like we specifically don’t have Jujutsu depending on OpenSSL at all on Darwin in Nixpkgs, so I wonder why it’s getting pulled in.

Reply:
Yes, that looks like it might be an oversight especially since `libgit2` depends on `openssl`, which is actually why we depend on it at all (so we can in theory get rid of it *entirely* if we drop libgit2, I think!) So the nixpkgs expression might just be a bit wrong.